qcodes/dataset/measurements.py

```diff
@@ -174,9 +174,10 @@ def add_result(self, *res_tuple: res_type) -> None:
                 inserting_as_arrays = True
                 inserting_this_as_array = True
             if any(isinstance(value, typ) for typ in array_like_types):
-
                 value = cast(np.ndarray, partial_result[1])
                 value = np.atleast_1d(value)
+                # we want to insert the casted version
```

Comment:
it's not only casted but also converted, right? for example, if `partial_result[1]` is a tuple, the tuple gets also converted to numpy array by `np.atleast_1d`? or does `cast(np.ndarray, ...` do that already?

Reply:
Converting from one type to another is a cast. However the function `cast` is indeed badly named and does not perform any real cast but purely changes the nominal type. The real `cast` is indeed performed by `atleast1d` as you say

Reply:
thanks, good to know. i got pythonized too much))